src/components/CreateCourse/components/Authors/Authors.jsx

```diff
@@ -0,0 +1,212 @@
+import { useContext, useRef, useState } from 'react';
+
+import classes from './Authors.module.css';
+
+import {
+	ADD_AUTHOR_BUTTON_TEXT,
+	ADD_AUTHOR_TITLE_TEXT,
+	AUTHOR_NAME_ERROR_MESSAGE,
+	AUTHOR_NAME_LABEL_TEXT,
+	AUTHORS_TITLE_TEXT,
+	COURSE_AUTHORS_TITLE_TEXT,
+	CREATE_AUTHOR_BUTTON_TEXT,
+	DELETE_AUTHOR_BUTTON_TEXT,
+	DURATION_ERROR_MESSAGE,
+	DURATION_LABEL_TEXT,
+	DURATION_TITLE_TEXT,
+	EMPTY_AUTHORS_lIST_MESSAGE,
+	ENTER_AUTHOR_NAME_PLACEHOLDER_TEXT,
+	ENTER_DURATION_IN_MINUTES_PLACEHOLDER_TEXT,
+} from '../../../../constants';
+
+import Input from '../../../../common/Input/Input';
+import Button from '../../../../common/Button/Button';
+
+import { pipeDuration } from '../../../../helpers/pipeDuration';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import { AuthorContext } from '../../../../GlobalContext/GlobalContext';
+import { CourseContext } from '../../CreateCourse';
+
+const Authors = () => {
+	const { duration, setDuration, courseAuthors, setCourseAuthors } =
+		useContext(CourseContext);
+	const { authors, setAuthors } = useContext(AuthorContext);
+
+	const [authorName, setAuthorName] = useState('');
+	const [isPristine, setIsPristine] = useState(true);
+	const [showAuthorError, setShowAuthorError] = useState(false);
+	const [showDurationError, setShowDurationError] = useState(false);
+	const [authorList, setAuthorList] = useState([...authors]);
+
+	const authorInput = useRef();
+
+	const changeDuration = (e) => {
+		const target = e.target;
+		const value = e.nativeEvent.data;
+
+		if (value < '0' || value > '9') {
+			target.value = duration || '';
+			return;
+		}
+
+		if (target.value === '') {
+			setDuration(0);
+
+			if (!showDurationError) {
+				setShowDurationError(true);
+			}
+
+			return;
+		}
+
+		if (+target.value === 0) {
+			if (!showDurationError) {
+				setShowDurationError(true);
+			}
+		} else {
+			if (showDurationError) {
+				setShowDurationError(false);
+			}
+		}
+
+		setDuration(+target.value);
```

Comment:
It is a bad habit to cast types like this, use `Number(<some string>)`. And I think that logical statement can be shortened as I commented higher